File: 프로그래머스/0/120863. 다항식 더하기/다항식 더하기.py
def solution(polynomial):
    a = 0
    b = 0
    polynomial_x = polynomial.split(" + ")
    for i in polynomial_x:
        if i.isnumeric():
            b += int(i)
        else:
            i_x = i.replace("x","")
            if i_x == "":
                a += 1
            else:
                a += int(i_x)
    if a == 0:
        answer = str(b)
    elif b == 0:
        answer = f"{a}x"
        if a == 1:
            answer = "x"
    else:
        answer = f"{a}x + {b}"
        if a == 1:
            answer = f"x + {b}"
        
    return answer
    

File: 프로그래머스/0/120863. 다항식 더하기/다항식 더하기.java
class Solution {
    public String solution(String polynomial) {
    int xn = 0;
    int n = 0;

    for (String term : polynomial.split(" \\+ ")) {
        if (term.contains("x")) {
            // "x" 단독 처리
            if (term.equals("x")) {
                xn += 1;
            } else {
                xn += Integer.parseInt(term.replace("x", ""));
            }
        } else {
            n += Integer.parseInt(term);
        }
    }

    // 결과 조합
    StringBuilder result = new StringBuilder();
    if (xn != 0) {
        if (xn == 1) result.append("x");
        else result.append(xn).append("x");
    }
    if (n != 0) {
        if (result.length() > 0) result.append(" + ");
        result.append(n);
    }

    return result.toString();
}



}